docs/user/install-gke-external.md

```diff
@@ -0,0 +1,229 @@
+
+You will then need to generate a cookie secret for later use:
+
+```
+python3 -c 'import os,base64; print(base64.urlsafe_b64encode(os.urandom(32)).decode())'
+```
+
+With the Turbinia repository cloned to your local machine, cd into the directory we'll be working from:
+
+```
+wyassine@wyassine:~/turbinia$ cd k8s/tools/
+```
+
+Then copy over the `oauth2_proxy.cfg` template via:
+
+```
+wyassine@wyassine:~/turbinia/k8s/tools$ cp ../../docker/oauth2_proxy/oauth2_proxy.cfg .
+```
+
+You will then need to edit the `oauth2_proxy.cfg` file and replace the following:
+
+- `CLIENT_ID`: The web client id
+- `CLIENT_SECRET`: The web client secret
+- `OIDC_EXTRA_AUDIENCES`: The native client id
+- `UPSTREAMS`: The domain name registered, ex: `upstreams = ['https://<DOMAIN>]`
+- `REDIRECT_URL`: The redirect URI you registered ex: `https://<DOMAIN>/oauth2/callback`
+- `COOKIE_SECRET`: The cookie secret you generated above
+- `EMAIl_DOMAINS`: The email domain name you'd allow to authenticate ex: `yourcompany.com`
+
+Run the following command to base64 encode the config file:
+
+```
+base64 -w0 oauth2_native.cfg > oauth2_native.b64
+```
+
+Then to deploy the config to the cluster
+
+```
+kubectl create configmap oauth2-config --from-file=OAUTH2_CONF=oauth2_native.b64
+```
+
+Create a file named `auth.txt` in your working directory and append a list of emails
+you'd like to allow access to the Turbinia app, one email per line. Once complete base64 encode:
+
+```
+base64 -w0 auth.txt > auth.b64
+```
+
+Then deploy the config to the cluster
+
+```
+kubectl create configmap auth-config --from-file=OAUTH2_AUTH_EMAILS=auth.b64
+```
+
+Lastly, deploy the Oauth2 Proxy to the cluster through the following command:
+
+```
+kubectl create -f ../celery/turbinia-oauth2-proxy.yaml
+```
+
+### 4. Deploy the Load Balancer and Managed SSL
+
+In the final step, edit `turbinia-ingress.yaml` located in the `k8s/celery` directory
+and replace the two placeholders `<DOMAIN>` with the domain you configured. Save
+the file then deploy it to the cluster:
+
+```
+kubectl create -f ../celery/turbinia-ingress.yaml
+```
+
+Within 10 minutes all the load balancer components should be ready and you should
+be able to externally connect to the domain name you configured. Additionally, you can check on the status of the load balancer via:
+
+```
+kubectl describe ingress turbinia-ingress
+```
+
+Congrats, you have now successfully configured Turbinia to be externally accessible!
+
+## Making Turbinia processing requests
+
+Once Turbinia is externally accessible, install the Turbinia client and Oauth
+Desktop credentials locally on your machine.
+
+- To create a processing request for evidence run the following:
+
+```
+turbinicatl googleclouddisk -d <DISK_NAME> -z <ZONE>
+```
+
+- To access the Turbinia Web UI, point your browser to:
+
+```
+https://<DOMAIN_YOU_CONFIGURED>
+```
+
+## Additional
+
+### Configuring an ipv6 address
```

Comment:
Should we link to this from the DNS section above too?

Reply:
ack